src/app/rosetta/lib/network.ml

```diff
@@ -232,18 +234,34 @@ module Status = struct
       { gql= (fun () -> Graphql.query (Get_status.make ()) graphql_uri)
       ; db_oldest_block=
           (fun () ->
-            Errors.Lift.sql ~context:"Oldest block query"
-            @@ Db.find oldest_block_query () )
+            match !oldest_block_ref with
+            | Some oldest_block -> Deferred.Result.return oldest_block
+            | None ->
+                let%map result =
+                  Errors.Lift.sql ~context:"Oldest block query"
+                  @@ Db.find oldest_block_query ()
+                in
+                Result.iter result ~f:(fun oldest_block -> oldest_block_ref := Some oldest_block) ;
+                result )
       ; validate_network_choice= Validate_choice.Real.validate }
   end
 
   module Impl (M : Monad_fail.S) = struct
     let handle ~graphql_uri ~(env : 'gql Env.T(M).t) (network : Network_request.t) =
       let open M.Let_syntax in
-      let%bind res = env.gql () in
+      let time operation_name f =
+        let start_time = Time.now () in
+        let%map x = f () in
+        let end_time = Time.now () in
+        let elapsed_time = Time.diff end_time start_time |> Time.Span.to_ms in
+        Core.Printf.printf "operation %s took %fms\n%!" operation_name elapsed_time ;
```

Comment:
I assume this is for debugging, and won't be in the final code.

Reply:
Yes, I will remove them before merging.